### PR TITLE
[de] Update list of versions to match English

### PR DESCRIPTION
**Pull request by AbdelatifAitBara**
Align https://kubernetes.io/de/docs/home/supported-doc-versions/ with https://kubernetes.io/docs/home/supported-doc-versions/

[Preview of the updated page.](https://deploy-preview-46350--kubernetes-io-main-staging.netlify.app/de/docs/home/supported-doc-versions/)

/language de

### CODE REVIEW

File: data/i18n/de/de.toml [hunk 1]
```diff
@@ -1,5 +1,14 @@
 # i18n strings for the German (main) site.
 
+[docs_version_latest_heading]
+other = "Aktuelle Version"
+
+[docs_version_other_heading]
+other = "Vorherige Versionen"
+
+[docs_version_current]
```
Comment (ricardoamaro) [inline data/i18n/de/de.toml:9]: The order this is presented in the other toml files for other languages is eg.:

```
data/i18n/en/en.toml:116:[docs_version_current]
data/i18n/en/en.toml-117-other = "(this documentation)"
data/i18n/en/en.toml-118-
data/i18n/en/en.toml:119:[docs_version_latest_heading]
data/i18n/en/en.toml-120-other = "Latest version"
data/i18n/en/en.toml-121-
data/i18n/en/en.toml:122:[docs_version_other_heading]
data/i18n/en/en.toml-123-other = "Older versions"
data/i18n/en/en.toml-124-
```
We might want to keep the same order

Comment (sftim) [reply data/i18n/de/de.toml:9]: Please order the localization strings based on sorting them using English collation rules.

Comment (AbdelatifAitBara) [reply data/i18n/de/de.toml]: Done.

**Pull request merged**